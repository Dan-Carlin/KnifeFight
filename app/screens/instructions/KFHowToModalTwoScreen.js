/*
KFHowToModalTwoScreen - View for second screen of instructions modal.
*/

// External libraries
import React from "react";
import { View } from "react-native";

// Assets
import {
  ArrowLeftGraphic,
  ArrowRightGraphic,
} from "../../assets/buttons/actions";
import { DiagramBGraphic } from "../../assets/diagrams";

// Components
import OpacityButton from "../../components/OpacityButton";
import Text from "../../components/Text";

// Resources
import strings from "../../config/strings";
import styles from "./KFHowToModalTwoStyles";

function KFHowToModalTwoScreen({ onPreviousPress, onNextPress }) {
  return (
    <View style={styles.screenContainer}>
      <View style={styles.bodyContainer}>
        <View style={styles.instructionsContainer}>
          <View style={styles.numberContainer}>
            <Text>{strings.modal_number_two_label}</Text>
          </View>
          <View style={styles.textContainer}>
            <Text>
              <Text style={styles.subtitle}>
                {strings.modal_instructions_two_subtitle}
              </Text>
              <Text>{strings.modal_instructions_two_text_a}</Text>
              <Text style={styles.highlightText}>
                {strings.modal_instructions_two_highlight}
              </Text>
              <Text>{strings.modal_instructions_two_text_b}</Text>
            </Text>
          </View>
        </View>
        <View style={styles.imageContainer}>
          <DiagramBGraphic />
        </View>
      </View>
      <View style={styles.buttonsContainer}>
        <View style={styles.button}>
          <OpacityButton Graphic={ArrowLeftGraphic} onPress={onPreviousPress} />
        </View>
        <View style={styles.pageLabelContainer}>
          <Text style={styles.pageLabel}>{strings.modal_page_two_label}</Text>
        </View>
        <View style={styles.button}>
          <OpacityButton Graphic={ArrowRightGraphic} onPress={onNextPress} />
        </View>
      </View>
    </View>
  );
}

export default KFHowToModalTwoScreen;
