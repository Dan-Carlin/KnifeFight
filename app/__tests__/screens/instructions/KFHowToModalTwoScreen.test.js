import React from "react";
import renderer from "react-test-renderer";

import KFHowToModalTwoScreen from "../../../screens/instructions/KFHowToModalTwoScreen";

describe("How To Modal second screen tests", () => {
  it("renders second How To screen correctly", () => {
    const tree = renderer.create(<KFHowToModalTwoScreen />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
