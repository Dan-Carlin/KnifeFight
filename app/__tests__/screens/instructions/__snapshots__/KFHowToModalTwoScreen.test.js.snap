// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`How To Modal second screen tests renders second How To screen correctly 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "flex": 0.85,
      }
    }
  >
    <View
      style={
        Object {
          "flexDirection": "row",
        }
      }
    >
      <View
        style={
          Object {
            "marginEnd": 6,
          }
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#fff",
                "fontFamily": "Avenir",
                "fontSize": 16,
                "width": "100%",
              },
              undefined,
            ]
          }
        >
          2.
        </Text>
      </View>
      <View
        style={
          Object {
            "flex": 1,
            "flexWrap": "wrap",
          }
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#fff",
                "fontFamily": "Avenir",
                "fontSize": 16,
                "width": "100%",
              },
              undefined,
            ]
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#fff",
                  "fontFamily": "Avenir",
                  "fontSize": 16,
                  "width": "100%",
                },
                Object {
                  "fontWeight": "bold",
                },
              ]
            }
          >
            During your turn: 
          </Text>
          <Text
            style={
              Array [
                Object {
                  "color": "#fff",
                  "fontFamily": "Avenir",
                  "fontSize": 16,
                  "width": "100%",
                },
                undefined,
              ]
            }
          >
            Use the tools screen to keep track of your remaining HP, handle dice rolls, and customize how your name will display to your rivals. Once finished, tap the 
          </Text>
          <Text
            style={
              Array [
                Object {
                  "color": "#fff",
                  "fontFamily": "Avenir",
                  "fontSize": 16,
                  "width": "100%",
                },
                Object {
                  "color": "#f26e85",
                  "fontStyle": "italic",
                  "fontWeight": "bold",
                },
              ]
            }
          >
            End Turn
          </Text>
          <Text
            style={
              Array [
                Object {
                  "color": "#fff",
                  "fontFamily": "Avenir",
                  "fontSize": 16,
                  "width": "100%",
                },
                undefined,
              ]
            }
          >
             button.
          </Text>
        </Text>
      </View>
    </View>
    <View
      style={
        Object {
          "height": "54%",
          "justifyContent": "flex-start",
          "padding": 10,
        }
      }
    >
      <SvgMock />
    </View>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 0.15,
        "flexDirection": "row",
        "justifyContent": "space-between",
      }
    }
  >
    <View
      style={
        Object {
          "height": "80%",
          "width": "25%",
        }
      }
    >
      <View
        accessible={true}
        focusable={false}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "height": "100%",
            "opacity": 1,
            "width": "100%",
          }
        }
      >
        <SvgMock />
      </View>
    </View>
    <View
      style={
        Object {
          "width": "50%",
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#fff",
              "fontFamily": "Avenir",
              "fontSize": 16,
              "width": "100%",
            },
            Object {
              "fontSize": 24,
              "fontWeight": "bold",
              "textAlign": "center",
            },
          ]
        }
      >
        2/3
      </Text>
    </View>
    <View
      style={
        Object {
          "height": "80%",
          "width": "25%",
        }
      }
    >
      <View
        accessible={true}
        focusable={false}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "height": "100%",
            "opacity": 1,
            "width": "100%",
          }
        }
      >
        <SvgMock />
      </View>
    </View>
  </View>
</View>
`;
